### PR TITLE
ui: Move ember-data classes to use native JS classes/decorators

### PR DESCRIPTION
**Pull request by johncowen**
This PR converts all our ember-data classes (Models, Adapters, Serializers) to use native JS classes plus decorators.

There are a couple of extra bits I did whilst going through, mainly just clean up, so I'll see if I can add a few inline comments to help there.

### CODE REVIEW

File: ui/packages/consul-ui/app/adapters/http.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { inject as service } from '@ember/service';
-import Adapter from 'ember-data/adapter';
-import AdapterError from '@ember-data/adapter/error';
+import Adapter from '@ember-data/adapter';
```
Comment (johncowen) [inline ui/packages/consul-ui/app/adapters/http.js:2]: Everything now uses the `@ember-data` org

File: ui/packages/consul-ui/app/models/node.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-  Node: attr('string'),
-  Meta: attr(),
-  Services: attr(),
-  Checks: attr(),
-  CreateIndex: attr('number'),
-  ModifyIndex: attr('number'),
-  TaggedAddresses: attr(),
-  Datacenter: attr('string'),
-  Segment: attr(),
-  Coord: attr(),
```
Comment (johncowen) [inline ui/packages/consul-ui/app/models/node.js:21]: I took out Coord and Segment from here, they seem to have been mistakenly added here from a long long time ago.

**Review by johncowen — COMMENTED** [review]
📔

**Review by kaxcode — APPROVED** [review]
LGTM 🎉

**Pull request merged**